src/GridRegistry.ts

```diff
@@ -0,0 +1,47 @@
+import { WidgetBaseConstructor } from '@dojo/widget-core/interfaces';
+import WidgetRegistry from '@dojo/widget-core/WidgetRegistry';
+import Body, { BodyInterface } from './Body';
+import Cell, { CellInterface } from './Cell';
+import Footer, { FooterInterface } from './Footer';
+import Header, { HeaderInterface } from './Header';
+import HeaderCell, { HeaderCellInterface } from './HeaderCell';
+import Row, { RowInterface } from './Row';
+
+export interface GridRegistered {
+	[key: string]: WidgetBaseConstructor;
+	body: BodyInterface;
+	cell: CellInterface;
+	footer: FooterInterface;
+	header: HeaderInterface;
+	'header-cell': HeaderCellInterface;
+	row: RowInterface;
+}
+
+export default class GridRegistry<T extends GridRegistered = GridRegistered> extends WidgetRegistry {
+	private _overrides: WidgetRegistry = new WidgetRegistry();
+
+	constructor() {
+		super();
+
+		super.define('body', Body);
```

Comment:
I am confused by the indirection here... why does `super.define()` required?  Why is the `define()` here having some sort of "hidden" registry? It either feels like registry is not properly extensible to have named and _n_ widgets working together, which we should look at.